streamlit-app/src/image_handler.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+import typing as t
+import cv2
+
+from ultralytics import YOLO
+
+from PIL import Image
+from sklearn.cluster import KMeans
+
+class ImageHandler():
+    def __init__(self, model_path: str, img: np.ndarray) -> None:
+        """
+        Initializes the ImageHandler class.
+
+        Parameters:
+        - model_path (str): The path to the model file.
+        - img (np.ndarray): The input image.
+        """
+        self.model = YOLO(model_path)
+        self.img = img
+        # get model results list
+        self.results = self.model.predict(img)
```

Comment:
I would move the prediction step out of this class. IMO, `ImageHandler` should depend on prediction results, but not run the prediction itself (or know the trained model).
Rationale:
- prediction step is a separate concern from handling images and it has its own complexities (e.g. passing the right params to YOLO)
- It would be better to handle it in a layer above so that the results can also be available to other modules/classes in the pipeline, e.g. coordinates translation
- in general, running an expensive I/O operation in a class constructor is bad practice
 (this will also make the depe